### ISSUE TITLE
Error during start up of the IDE and sync process

### ISSUE THREAD

**Issue by sagarmj-cloud** [derkork/intellij-leiningen-plugin#34]
I get the following error when I start my intellij ide and the lein plugin throws the following error.

ull
java.lang.AssertionError
    at com.intellij.openapi.roots.impl.libraries.LibraryImpl.getFiles(LibraryImpl.java:164)
    at com.intellij.openapi.roots.impl.libraries.LibraryImpl$MyRootProviderImpl.getFiles(LibraryImpl.java:561)
    at com.intellij.openapi.roots.impl.LibraryOrderEntryBaseImpl.getRootFiles(LibraryOrderEntryBaseImpl.java:69)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl$IndexState.a(DirectoryIndexImpl.java:784)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl$IndexState.a(DirectoryIndexImpl.java:1045)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl$IndexState.access$2600(DirectoryIndexImpl.java:486)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl.d(DirectoryIndexImpl.java:419)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl.access$400(DirectoryIndexImpl.java:54)
    at com.intellij.openapi.roots.impl.DirectoryIndexImpl$4.rootsChanged(DirectoryIndexImpl.java:123)
    at sun.reflect.GeneratedMethodAccessor73.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:120)
    at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:228)
    at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:234)
    at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:219)
    at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:216)
    at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:209)
    at com.intellij.util.messages.impl.MessageBusImpl.access$000(MessageBusImpl.java:43)
    at com.intellij.util.messages.impl.MessageBusImpl$1.invoke(MessageBusImpl.java:131)
    at com.sun.proxy.$Proxy89.rootsChanged(Unknown Source)
    at com.intellij.openapi.roots.impl.ProjectRootManagerComponent.fireRootsChangedEvent(ProjectRootManagerComponent.java:207)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl.b(ProjectRootManagerImpl.java:408)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl.access$100(ProjectRootManagerImpl.java:55)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$BatchSession.a(ProjectRootManagerImpl.java:107)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$BatchSession.rootsChanged(ProjectRootManagerImpl.java:120)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl.mergeRootsChangesDuring(ProjectRootManagerImpl.java:319)
    at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$LibraryTableMultilistener.afterLibraryAdded(ProjectRootManagerImpl.java:510)
    at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:89)
    at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:34)
    at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:66)
    at com.sun.proxy.$Proxy36.afterLibraryAdded(Unknown Source)
    at com.intellij.openapi.roots.impl.libraries.LibraryTableBase.a(LibraryTableBase.java:116)
    at com.intellij.openapi.roots.impl.libraries.LibraryTableBase.a(LibraryTableBase.java:176)
    at com.intellij.openapi.roots.impl.libraries.LibraryTableBase.access$300(LibraryTableBase.java:38)
    at com.intellij.openapi.roots.impl.libraries.LibraryTableBase$LibraryModel.commit(LibraryTableBase.java:213)
    at de.janthomae.leiningenplugin.module.ModuleCreationUtils$5.run(ModuleCreationUtils.java:314)
    at com.intellij.openapi.application.RunResult.run(RunResult.java:37)
    at com.intellij.openapi.application.WriteAction$1.run(WriteAction.java:35)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:199)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:708)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:669)
    at java.awt.EventQueue$2.run(EventQueue.java:667)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:678)
    at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:700)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:525)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:348)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)

**Comment by sagarmj-cloud** [comment]
Could you kindly provide any help on this? When I start my intellij 12.1 ide and I have the lein plugin installed, I get the above stacktrack. 
The stack trace starts with null and then the rest is as above.
Any help is appreciated thank you.

**Comment by 256BitChris** [comment]
It's hard to say - do you have a sample project that you can attach to this issue that will reproduce the error?  You might also try blowing away your .idea folder and .iml files and starting over.

**Comment by derkork** [comment]
@sagarmj76  did resetting the project work?

**Comment by ChrisDevo** [comment]
@derkork I had the same problem. I deleted the .iml file and ran

``` shell
lein pom
mvn idea:idea
```

This seems to have fixed the problem.

**Comment by sagarmj-cloud** [comment]
thanks a lot for the suggestion. 

thanks
Sagar

**Comment by markmandel** [comment]
I just got this as well.

I got this after I attempted to import module into an existing project via a project.clj.

Hopefully you will be able to reproduce it that way.

**Comment by markmandel** [comment]
This issue keeps rearing it's head for me.  I kept getting it through these steps:
1. Import a project from a Leiningen project.clj
2. Import a new module from another Leiningen project.clj (In my case this is symlinked to my ./checkouts/ dir, not sure if that is a factor.
3. Close IntelliJ
4. Start IntelliJ back up and open the project. This exception occurs.

The only workaround I've found so far is to Invalidate the Cache and restart IntelliJ, and that seems to resolve it.  Otherwise my IntelliJ doesn't render properly and I can't use it.  But I have to do it every time.

My IntelliJ version is: 12.1.4, Ubuntu Linux 13.04, 64 Bit.

Let me know if I can provide any more help.
